authui/src/authflowv2/bodyScrollLock.ts

```diff
@@ -0,0 +1,26 @@
+import { Controller } from "@hotwired/stimulus";
+
+export class BodyScrollLockController extends Controller {
+  static targets = ["dialog"];
+
+  scrollPosition = 0;
+
+  lock() {
+    this.scrollPosition = document.documentElement.scrollTop;
+    document.body.style.overflow = "hidden";
+    document.body.style.position = "fixed";
+    document.body.style.top = `-${this.scrollPosition}px`;
+    document.body.style.width = "100%";
+    document.body.style.height = visualViewport?.height
+      ? `${visualViewport.height}px`
+      : "100vh";
+  }
+
+  unlock() {
+    document.body.style.removeProperty("overflow");
+    document.body.style.removeProperty("position");
+    document.body.style.removeProperty("top");
+    document.body.style.removeProperty("width");
```

Comment:
If body originally has some of these property defined, removing them will break things. Consider using insertRule() https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule

In lock(), we create a style element and append it to the body. In unlock(), we remove the style element we created in lock() from the body.

Reply:
Updated 🙏